plutus-tx/src/PlutusTx/ErrorCodes.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+module PlutusTx.ErrorCodes where
+
+import           PlutusTx.Builtins as Builtins
+
+{-
+All error codes in this module should be unique and can be used only once!
+They help to trace the errors in on-chain code.
+-}
+
+-- | The error happens in TH generation of indexed data
+{-# INLINABLE reconstructCaseError #-}
+reconstructCaseError :: Builtins.BuiltinString
+reconstructCaseError = "PT01"
```

Comment:
suggest dropping the leading 0: that suggests it's only ever going to be a 2-digit number, but who knows D: